mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/MantisJobMetadataImpl.java

```diff
@@ -240,18 +240,19 @@ boolean replaceWorkerMetaData(int stageNum,
      public boolean addWorkerMetadata(int stageNum, JobWorker newWorker)
             throws InvalidJobException {
          if(logger.isTraceEnabled()) { logger.trace("Adding workerMetadata {} for stage {}", stageNum, newWorker); }
-        boolean result=true;
         if(!stageMetadataMap.containsKey(stageNum)) {
         	logger.warn("No such stage {}", stageNum);
         }
-        if(!((MantisStageMetadataImpl)stageMetadataMap.get(stageNum)).addWorkerIndex(newWorker))
-            result=false;
-        Integer integer = workerNumberToStageMap.put(newWorker.getMetadata().getWorkerNumber(), stageNum);
-        if(integer != null && integer != stageNum) {
-            logger.error(String.format("Unexpected to put worker number mapping from %d to stage %d for job %s, prev mapping to stage %d",
-            		newWorker.getMetadata().getWorkerNumber(), stageNum, newWorker.getMetadata().getJobId(), integer));
+        final boolean result = ((MantisStageMetadataImpl) stageMetadataMap.get(stageNum)).addWorkerIndex(newWorker);
+
+        if (result) {
+            Integer integer = workerNumberToStageMap.put(newWorker.getMetadata().getWorkerNumber(), stageNum);
```

Comment:
This is a behavior change where we aren't updating the map when `result = false`. Would `workerNumberToStageMap` have the mapping for an existing worker esp. during leader election or bootstrap from state?

Reply:
Yes, this is behavior change. But the thinking here is that this is more correct than the existing logic because if you cannot add the worker to the stage, then the workerNumberToStageMap should also reflect that.

Reply:
Let me know if the rationale here makes sense.